File: Pruebas/U4/Alumno.java

public class Alumno extends Persona {
  private Notas misnotas;
  public Alumno(String nombre, String rut, int cantidadNotas) {
    super(nombre);
    this.asignaRUT(rut);
    this.validaRUT();
    this.misnotas = new Notas(cantidadNotas);
  }
  public void agregaNota(int i, float nota) {
	  this.misnotas.addNota(i, nota);
  }
  public float promedio() {
	  return (misnotas.promedio());
  }
  public String libroClases() {
	  return( this.obtieneDatos() + " " + this.misnotas.diceNotas());
	  
  }
}

File: Pruebas/U4/PrincipalPHPU4/Alumno.php
<?php
include("Persona.php");
include("Notas.php");
class Alumno extends Persona
{
    
    private Notas $misnotas;
    
    public function __construct(string $nom, string $rut, int $cantidad)
    {
        $this->nombre = $nom;
        $this->asignaRut($rut);
        $this->validaRut();
        $this->misnotas = new Notas($cantidad);
    }
    
    public function libroclases(){
        return $this->obtieneDatos() . " " . $this->misnotas->diceNotas();
    }
    
    public function agregarnotas(int $i, float $nota){
        $this->misnotas->addNota($i, $nota);
    }
    
    public function promedio(){
        return $this->misnotas->promedio();
    }
}

